addons/account/tests/test_account_move_reconcile.py

```diff
@@ -4438,3 +4384,179 @@ def test_reconcile_payment_custom_rate(self):
         lines_to_reconcile.reconcile()
 
         self.assertTrue(all(lines_to_reconcile.mapped('reconciled')), "All lines should be fully reconciled")
+
+    def test_cash_basis_with_analytic_distribution(self):
+        """
+        Check that the analytic distribution is applied correctly to the cash basis move lines.
+        """
+        self.env.company.tax_exigibility = True
+
+        analytic_plan = self.env['account.analytic.plan'].create({
+            'name': 'Default',
+            'company_id': False
+        })
+        analytic_account_a = self.env['account.analytic.account'].create({
+            'name': 'analytic_account_a',
+            'plan_id': analytic_plan.id,
+            'company_id': False,
+        })
+        analytic_account_b = self.env['account.analytic.account'].create({
+            'name': 'analytic_account_b',
+            'plan_id': analytic_plan.id,
+            'company_id': False,
+        })
+        analytic_distribution_a = {
+            analytic_account_a.id: 100,
+        }
+        analytic_distribution_b = {
+            analytic_account_b.id: 100,
+        }
+        analytic_distribution_a_serialized = {
+            str(analytic_account_a.id): 100,
+        }
+        analytic_distribution_b_serialized = {
+            str(analytic_account_b.id): 100,
+        }
+
+        tax = self.env['account.tax'].create({
+            'name': 'cash basis 20%',
+            'type_tax_use': 'purchase',
+            'amount': 20,
+            'tax_exigibility': 'on_payment',
+            'cash_basis_transition_account_id': self.cash_basis_transfer_account.id,
+            'invoice_repartition_line_ids': [
+                (0, 0, {
```

Comment:
You use Command.create below, but not in this block ; why ? :thinking: Should be everywhere.